### PR TITLE
add display post widget into frontend

### PR DESCRIPTION
**Pull request by btn114**
### Fixed Issues (if relevant)

1. https://github.com/mageplaza/magento-2-blog/<issue_number>: Issue title
2. ...

### Manual testing scenarios

1. 
2. 

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)

### CODE REVIEW

File: Block/Widget/Posts.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Created by PhpStorm.
```
Comment (isaacdecuba) [inline Block/Widget/Posts.php]: I think it's a lot nicer to remove this comment tag and add a phpdoc right before the class name

File: Block/Widget/Posts.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+namespace Mageplaza\Blog\Block\Widget;
+
+use Magento\Widget\Block\BlockInterface;
+use Mageplaza\Blog\Block\Frontend;
+class Posts extends Frontend implements BlockInterface{
+
+	protected $_template = "widget/posts.phtml";
+
+	public function getCollection(){
+		if ($this->hasData('show_type') && $this->getData('show_type')=='category')
```
Comment (isaacdecuba) [inline Block/Widget/Posts.php]: Not bad, `$this->getData('show_type') === 'category'` would be nicer

File: Block/Widget/Posts.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		{
+			$postsCollection=$this->helperData->categoryfactory->create()->load($this->getData('category_id'))->getSelectedPostsCollection();
+		}
+		else {
+			$postsCollection = $this->helperData->postfactory->create()->getCollection();
+		}
+		$postsCollection
+			->addOrder('created_at')
+			->setPageSize($this->getData('post_count'))
+		;
```
Comment (isaacdecuba) [inline Block/Widget/Posts.php]: please move `;` onto previous line

File: view/frontend/templates/widget/posts.phtml [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+?>
+<?php
+$posts = $block->getCollection();
+?>
+
+
+    <div class="post-list-container mpcss">
+        <div class="row">
+            <?php if ($block->getBlogConfig('sidebar/sidebar_left_right')==1){
+				?>
```
Comment (isaacdecuba) [inline view/frontend/templates/widget/posts.phtml]: - Php closing tag should go onto previous line.
- I believe that getBlogConfig will return a string, if so please change line to `$block->getBlogConfig('sidebar/sidebar_left_right') === '1'`

File: view/frontend/templates/widget/posts.phtml [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+
+                                <div class="mp-post-info">
+                                    <?php echo __('Posted on') . $block->formatCreatedAt($post->getCreatedAt()); ?> |
+									<?php echo __('Posted in') . $block->getPostCategoryHtml($post); ?>
+                                </div>
+
+
+                                <div class="post-short-description">
+                                    <p><?php echo $post->getShortDescription(); ?></p>[...]
```
Comment (isaacdecuba) [inline view/frontend/templates/widget/posts.phtml]: Missing `escapeHtml()`

File: view/frontend/templates/widget/posts.phtml [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+                                </div>
+
+                                <div class="mp-clearfix"></div>
+
+
+                                <div class="mp-post-meta-info mp-alignright">
+                                        <div class="actions">
+                                            <a class="mp-read-more" href="<?php echo $block->getUrlByPost($post); ?>"
+                                               title="<?php echo __('Read More'); ?>"><?php echo __('Read more'); ?></a>
```
Comment (isaacdecuba) [inline view/frontend/templates/widget/posts.phtml]: missing `escapeHtml`

File: Block/Widget/Posts.php (hunk 1 above)
Comment (imsamthomas) [inline Block/Widget/Posts.php]: Thanks for suggestion.
We will replace this.

**Pull request merged**